src/main/java/uk/gov/pay/products/service/PaymentCreator.java

```diff
@@ -145,7 +146,10 @@ private PaymentEntity mergePaymentEntityWithReferenceNumberCheck(PaymentEntity p
                 
                 if (PanDetector.isSuspectedPan(paymentEntity.getReferenceNumber())) {
                     logger.warn("Suspected PAN entered by user in reference field",
-                            kv(PAYMENT_EXTERNAL_ID, paymentEntity.getGovukPaymentId()));
+                            kv(PAYMENT_EXTERNAL_ID, paymentEntity.getGovukPaymentId()),
+                            kv(GATEWAY_ACCOUNT_ID, paymentEntity.getGatewayAccountId()),
+                            kv("reference_length", PanDetector.cleanedReference(paymentEntity.getReferenceNumber()).length())
```

Comment:
This isn’t quite the reference length, is it? It’s the length of the reference after it’s been “cleaned”. Since cleaning involves removing dashes and spaces, it could be shorter. Maybe call this `number_of_digits` or something?